### PR TITLE
Move PaginatedResponseHelper class to Commands.Common

### PR DESCRIPTION
**Pull request by NonStatic2014**
Paginated response from a web service is a common case. So the helper class to take care this situation should also be in a common place. This change is to move it from ResourceManager cmdlet to Commands.Common. Machine Learning cmdlet will consume it later.

All tests under Resource Manager cmdlet have been run locally. No failed cases (129 passed, 25 skipped).

### CODE REVIEW

**Comment by NonStatic2014** [comment]
On Demand passed: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1130/console

**Comment by markcowl** [comment]
@NonStatic2014 Given that this is code complete, and this shouldn't be required for the release, we are likely not going to take this until after the release.

**Comment by markcowl** [comment]
This will go in for next sprint - discussed a change with Klein

**Comment by NonStatic2014** [comment]
On-demand build passed: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1176/

File: src/Common/Commands.Common/Commands.Common.csproj [hunk 1]
```diff
@@ -183,4 +183,4 @@
     </ProjectReference>
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-</Project>
+</Project>
```
Comment (markcowl) [inline src/Common/Commands.Common/Commands.Common.csproj]: remove any changes here

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/CmdletBase/ResourceManagerCmdletBase.cs [hunk 2]
```diff
@@ -12,26 +12,26 @@
...
-    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkClient;
     using System.Threading;
     using System.Linq;
     using System.Management.Automation;
     using System.Runtime.ExceptionServices;
     using System.Threading.Tasks;
+    using Common;
+    using Microsoft.Azure.Commands.Common.Authentication;
+    using Microsoft.Azure.Commands.Common.Authentication.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Components;
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/CmdletBase/ResourceManagerCmdletBase.cs]: Why are you changing this file?

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Locations/GetAzureLocationCmdlet.cs [hunk 3]
```diff
@@ -14,19 +14,20 @@
...
-    using Microsoft.Azure.Management.ResourceManager.Models;
-    using Microsoft.WindowsAzure.Commands.Utilities.Common;
     using System;
     using System.Collections.Generic;
     using System.Linq;
     using System.Management.Automation;
+    using Microsoft.Azure.Commands.ResourceManager.Common;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Extensions;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels;
+    using Microsoft.Azure.Management.ResourceManager.Models;
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Locations/GetAzureLocationCmdlet.cs]: Why are you changing this file?

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Lock/GetAzureResourceLockCmdlet.cs [hunk 4]
```diff
@@ -14,11 +14,11 @@
 
 namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
 {
-    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Components;
-    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Extensions;
-    using Newtonsoft.Json.Linq;
     using System.Management.Automation;
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Lock/GetAzureResourceLockCmdlet.cs]: Why are you changign this file?

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/CmdletBase/ResourceManagerCmdletBase.cs [hunk 5]
```diff
@@ -12,26 +12,26 @@
...
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Entities.ErrorResponses;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Entities.Resources;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Extensions;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.RestClients;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkClient;
+    using Microsoft.WindowsAzure.Commands.Common;
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/CmdletBase/ResourceManagerCmdletBase.cs]: Microsoft.WindowsAure.Commands.Common should not be added.  Why change this file at all?

**Comment by markcowl** [comment]
@vivsriaus Can you sign off on the changes to resource manager cmdlets?

**Comment by markcowl** [comment]
@NonStatic2014 A couple of minor comments from me, checking with Vivek on the resource manager changes.

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/CmdletBase/ResourceManagerCmdletBase.cs (hunk 5 above)
Comment (NonStatic2014) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/CmdletBase/ResourceManagerCmdletBase.cs]: "using"s are re-ordered automatically by VS.

Comment (NonStatic2014) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/CmdletBase/ResourceManagerCmdletBase.cs]: Removed.

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/ApiVersionHelper.cs [hunk 6]
```diff
@@ -26,6 +22,11 @@ namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Components
     using System.Threading;
     using System.Threading.Tasks;
 
+    using Entities.Providers;
```
Comment (vivsriaus) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/ApiVersionHelper.cs]: Can you please sort the usings for all these files?

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/ApiVersionHelper.cs [hunk 7]
```diff
@@ -108,7 +109,7 @@ private static string[] GetApiVersionsForResourceType(AzureContext context, stri
                 ? "/providers"
                 : string.Format("/subscriptions/{0}/providers", defaultSubscription.Id);
 
-            var providers = PaginatedResponseHelper.Enumerate(
+            ResourceProviderDefinition[] providers = PaginatedResponseHelper.Enumerate(
```
Comment (vivsriaus) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/ApiVersionHelper.cs]: Why? I prefer var here, instead of the type. That's the convention we follow in the project

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/HttpClientHelper.cs [hunk 8]
```diff
@@ -63,7 +63,7 @@ public virtual HttpClient CreateHttpClient(params DelegatingHandler[] primaryHan
             {
                 new AuthenticationHandler(cloudCredentials: credentials),
                 new UserAgentHandler(headerValues: headerValues),
-                new CmdletInfoHandler(cmdletHeaderValues: cmdletHeaderValues),
+                new Handlers.CmdletInfoHandler(cmdletHeaderValues: cmdletHeaderValues),
```
Comment (vivsriaus) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/HttpClientHelper.cs]: Please revert this

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Extensions/IEnumerableExtensions.cs [hunk 9]
```diff
@@ -14,27 +14,17 @@
...
     using System.Linq;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Collections;
 
     /// <summary>
     /// IEnumerable extension methods
     /// </summary>
     public static class IEnumerableExtensions
     {
         /// <summary>
-        /// Coalesces the enumerable.
```
Comment (vivsriaus) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Extensions/IEnumerableExtensions.cs]: Why is this removed?

**Comment by shahabhijeet** [comment]
@NonStatic2014 please address code review feedback.

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/ApiVersionHelper.cs (hunk 6 above)
Comment (NonStatic2014) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/ApiVersionHelper.cs]: Visual Studio prefer "System.xxx" on top and then other using in alphabet order.

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/ApiVersionHelper.cs (hunk 7 above)
Comment (NonStatic2014) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/ApiVersionHelper.cs]: Before I made this change, VS build failed due to ambiguous type and then I have to make this change.

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Extensions/IEnumerableExtensions.cs (hunk 9 above)
Comment (NonStatic2014) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Extensions/IEnumerableExtensions.cs]: It is not removed, it is moved to Common library as well.

Comment (vivsriaus) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Extensions/IEnumerableExtensions.cs]: No, I see it being removed from IEnumerableExtensions.cs

**Comment by NonStatic2014** [comment]
On demand build passed: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1181

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Policy/PolicyDefinitionCmdletBase.cs [hunk 10]
```diff
@@ -14,12 +14,12 @@
 
 namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
 {
-    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Components;
-    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Extensions;
-    using Newtonsoft.Json.Linq;
     using System.Linq;
     using System.Management.Automation;
     using System.Threading.Tasks;
+    using Microsoft.Azure.Commands.ResourceManager.Common;
```
Comment (vivsriaus) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Policy/PolicyDefinitionCmdletBase.cs]: Please reorder the usings

Comment (vivsriaus) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Policy/PolicyDefinitionCmdletBase.cs]: Please reorder the usings

**Comment by vivsriaus** [comment]
@NonStatic2014 LGTM, pending one minor nit.
